runtime/convertValues.go

```diff
@@ -465,3 +500,78 @@ func importCompositeValue(
 		nil,
 	)
 }
+
+func importPublicKey(
+	inter *interpreter.Interpreter,
+	runtimeInterface Interface,
+	fields *interpreter.StringValueOrderedMap,
+) *interpreter.CompositeValue {
+
+	var publicKeyValue *interpreter.ArrayValue
+	var signAlgoValue *interpreter.CompositeValue
+
+	fields.Foreach(func(fieldName string, value interpreter.Value) {
+		switch fieldName {
+		case sema.PublicKeyPublicKeyField:
+			arrayValue, ok := value.(*interpreter.ArrayValue)
+			if !ok {
+				panic(fmt.Errorf(
+					"cannot import value of type '%s'. invalid value for field '%s': %v",
+					sema.PublicKeyType,
+					fieldName,
+					value,
+				))
+			}
+
+			publicKeyValue = arrayValue
+
+		case sema.PublicKeySignAlgoField:
+			compositeValue, ok := value.(*interpreter.CompositeValue)
+			if !ok {
+				panic(fmt.Errorf(
+					"cannot import value of type '%s'. invalid value for field '%s': %v",
+					sema.PublicKeyType,
+					fieldName,
+					value,
+				))
+			}
+
+			signAlgoValue = compositeValue
```

Comment:
Do we have to validate the composite value itself here too, or is this done before / after?

Reply:
same for the signature algorithm as well.